### ISSUE TITLE
Error with compiling source code using 'make'

### ISSUE THREAD

**Issue by sroseallen** [lpipes/AncestralClust#8]
Hello!

I seem to be having some trouble compiling the code using my Ubuntu setup; all commands are up-to-date, however running 'make' produces the below error:

```
/usr/bin/ld: /tmp/ccoySM24.o:(.bss+0x20): multiple definition of 'lock'; /tmp/ccmx67dU.o:(.bss+0x0): first defined here
collect2: error: ld returned 1 exit status
make: *** [Makefile:26: ancestralclust] Error 1
```

Admitedly I'm not familiar with C, so I'm not confident where the error is coming from. In case this isn't an issue with my set-up and if the code is still maintained, would it be possible to check that the source code files can still be complied?

Thanks in advance! :+1:
-S

**Comment by lpipes** [comment]
What version of Ubuntu do you have?

**Comment by sroseallen** [comment]
Hi there - apologies, I've just seen the comment on the commit to use Ubuntu v20.04, and have just compiled using this version instead and everything worked perfectly (I was using v22.04). Thanks for your help!

**Comment by hempelc** [comment]
Sorry to revive an old thread, but I've had the same issue. I'm on Ubuntu v23.10 and received the same error that @sroseallen stated above. I then found this thread and tried to run make via Ubuntu v20.04 through a Docker image, but then I got this error (I'm printing just the last line of the error, hopefully this is sufficient):
```
math.c:1129:91: note: include '<math.h>' or provide a declaration of 'log'
make: *** [Makefile:26: ancestralclust] Error 1
```
Any idea how to fix this? Also, are there any plans to containerize AncestralClust like you have done for tronko? I'm trying to run tronko for taxonomic annotation and was hoping to use AncestralClust to built custom reference databases.

Thanks!
Chris
